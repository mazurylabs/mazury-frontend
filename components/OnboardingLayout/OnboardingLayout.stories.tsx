import { ComponentMeta, ComponentStory } from '@storybook/react';
import { Button } from 'components';
import * as React from 'react';
import { useState } from 'react';

import { OnboardingLayout } from './OnboardingLayout';

export default {
  title: 'Components/OnboardingLayout',
  component: OnboardingLayout,
  argTypes: {
    firstHeading: { control: { type: 'text' } },
    secondHeading: { control: { type: 'text' } },
    bottomContent: { control: { type: 'text' } },
    children: { control: { type: 'text' } },
  },
} as ComponentMeta<typeof OnboardingLayout>;

const Template: ComponentStory<typeof OnboardingLayout> = (args) => (
  <OnboardingLayout {...args} />
);

export const Default = Template.bind({});
Default.args = {
  firstHeading: 'Welcome to your new account',
  secondHeading: 'wojtek.eth',
  children: (
    <div className="mt-6 flex flex-col">
      <h1>Hello</h1>
    </div>
  ),
};
