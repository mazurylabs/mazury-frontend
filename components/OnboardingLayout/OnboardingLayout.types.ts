import { ReactNode } from 'react';

export interface OnboardingLayoutProps {
  firstHeading: ReactNode;
  secondHeading: ReactNode;
  children: ReactNode;
  bottomButtonText?: ReactNode;
  bottomButtonOnClick?: () => Promise<void>;
  bottomButtonDisabled?: boolean;
  overrideOnClick?: boolean;
}
