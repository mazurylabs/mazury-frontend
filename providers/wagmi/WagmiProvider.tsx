import * as React from 'react';
// import {
//   createClient,
//   configureChains,
//   defaultChains,
//   WagmiConfig,
// } from 'wagmi';
import { Connector, Provider, chain, defaultChains } from 'wagmi';
// import { infuraProvider } from 'wagmi/providers/infura';
// import { publicProvider } from 'wagmi/providers/public';
// import { jsonRpcProvider } from 'wagmi/providers/jsonRpc';
import { providers } from 'ethers';
import { InjectedConnector } from 'wagmi/connectors/injected';
import { WalletConnectConnector } from 'wagmi/connectors/walletConnect';

// const { chains, provider, webSocketProvider } = configureChains(defaultChains, [
//   infuraProvider(),
//   jsonRpcProvider({
//     rpc: (chain) => ({
//       http: `https://rpc.ankr.com/polygon_mumbai`,
//     }),
//   }),
//   publicProvider(),
// ]);

// const connectors = [
//   new InjectedConnector({ chains }),
//   new WalletConnectConnector({
//     chains,
//     options: {
//       qrcode: true,
//     },
//   }),
// ];

const chains = defaultChains;
const defaultChain = chain.polygonTestnetMumbai;

type ConnectorsConfig = { chainId?: number };
const connectors = ({ chainId }: ConnectorsConfig) => {
  return [
    new InjectedConnector({ chains }),
    new WalletConnectConnector({
      chains,
      options: {
        qrcode: true,
      },
    }),
  ];
};

// const client = createClient({
//   autoConnect: true,
//   provider,
//   webSocketProvider,
//   connectors,
// });

const isDev = process.env.NODE_ENV;
const providerEndpoint = isDev
  ? process.env.NEXT_PUBLIC_POLYGON_MUMBAI_URL
  : process.env.NEXT_PUBLIC_POLYGON_URL;

const provider = new providers.JsonRpcProvider(providerEndpoint);

// autoConnect

export const WagmiProvider: React.FC = ({ children }) => (
  <Provider connectors={connectors} provider={provider}>
    {children}
  </Provider>
);
