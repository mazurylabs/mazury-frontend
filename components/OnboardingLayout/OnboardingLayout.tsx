import { Button } from 'components';
import { OnboardingContext } from 'contexts';
import { useProfile } from 'hooks';
import Image from 'next/image';
import { useRouter } from 'next/router';
import { FC, useContext, useEffect } from 'react';
import { useAccount, useConnect } from 'wagmi';
import { OnboardingLayoutProps } from './OnboardingLayout.types';

const onboardingRoutes = [
  '/',
  'role',
  'refer',
  // 'write', ** we have removed this page from the onboarding flow because it is now accessible via the individual 'refer' buttons
  'whitelist',
  'socials',
  'finish',
  // 'redirect', we dont need a hard redirect anymore. we'll just do router.push('people/[address]')
];

export const OnboardingLayout: FC<OnboardingLayoutProps> = ({
  children,
  firstHeading,
  secondHeading,
  bottomButtonText = 'CONTINUE',
  bottomButtonOnClick,
  bottomButtonDisabled = false,
  overrideOnClick = false,
}) => {
  const router = useRouter();
  const [{ data: connectData, loading: connectLoading }] = useConnect();
  const [{ data: accountData }] = useAccount();
  const { profile: profileData } = useProfile(accountData?.address as string);
  const {
    setFormData,
    fetched,
    setFetched,
    setTwitterConnected,
    setGithubConnected,
  } = useContext(OnboardingContext);

  const goForward = async () => {
    const currentRoute = [...router.pathname.split('/'), '/'][2];
    if (bottomButtonOnClick) {
      await bottomButtonOnClick();
    }
    const nextRoute =
      onboardingRoutes[onboardingRoutes.indexOf(currentRoute) + 1];
    router.push(`/onboarding/${nextRoute}`);
  };

  const goBack = () => {
    if (router.pathname === '/onboarding') {
      return router.push('/');
    }
    const currentRoute = router.pathname.split('/')[2];
    const prevRoute =
      onboardingRoutes[onboardingRoutes.indexOf(currentRoute) - 1];
    router.push(`/onboarding/${prevRoute}`);
  };

  useEffect(() => {
    // console.log({ twitter: profileData?.twitter });
    if (profileData && !fetched) {
      const {
        bio,
        username,
        role_community_manager,
        role_creator,
        role_designer,
        role_developer,
        role_investor,
        role_researcher,
        role_trader,
        open_to_opportunities,
        email,
        website,
        ens_name,
        eth_address,
        avatar,
        twitter,
        github,
      } = profileData;
      setFormData({
        bio,
        username,
        role_community_manager,
        role_creator,
        role_designer,
        role_developer,
        role_investor,
        role_researcher,
        role_trader,
        open_to_opportunities,
        email,
        website,
        ens_name,
        eth_address,
        avatar,
        twitter,
        github,
      });
      if (profileData.twitter) {
        setTwitterConnected(true);
      }
      if (profileData.github) {
        setGithubConnected(true);
      }
      setFetched(true);
    }
  }, [
    profileData,
    setFormData,
    fetched,
    setFetched,
    setTwitterConnected,
    setGithubConnected,
  ]);

  if (connectLoading) {
    return <div>Loading...</div>;
  }

  if (!connectData?.connected) {
    return <div>Please connect your wallet first</div>;
  }

  return (
    <main className="mx-auto flex min-h-screen flex-col px-4 py-4 align-top sm:w-[530px]">
      <div className="absolute top-6 left-4 hidden w-full xl:flex">
        {/* next/Image overrides display:hidden */}
        {/* eslint-disable-next-line @next/next/no-img-element */}
        <img
          src="/waves.png"
          className="rounded-full hover:cursor-pointer"
          alt="Mazury logo"
          width="40px"
          height="40px"
          onClick={() => router.push('/')}
        />
      </div>

      <div className="xl:mt-20">
        <Image
          onClick={goBack}
          src="/icons/back.svg"
          alt="Back"
          width="16px"
          height="16px"
          layout="fixed"
          className="hover:cursor-pointer"
        />
      </div>

      <h1 className="mt-6 text-sm font-medium uppercase text-indigoGray-40">
        {firstHeading}
      </h1>
      <h3 className="mt-2 font-demi text-4xl text-indigoGray-90">
        {secondHeading}
      </h3>

      {children}

      <div className="mx-auto mb-8 mt-8 w-full">
        <Button
          size="large"
          onClick={overrideOnClick ? bottomButtonOnClick : goForward}
          className="w-full justify-center uppercase"
          disabled={bottomButtonDisabled}
        >
          {bottomButtonText}
        </Button>
      </div>
    </main>
  );
};
