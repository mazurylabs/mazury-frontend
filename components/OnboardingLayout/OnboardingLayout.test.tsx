import { render, screen } from '@testing-library/react';
import { OnboardingLayout } from './OnboardingLayout';
import * as wagmi from 'wagmi';

describe('OnboardingLaout', () => {
  test('all sections are being rendered', () => {
    const spyUseConnect = jest.spyOn(wagmi, 'useConnect');
    spyUseConnect.mockReturnValue([
      {
        // @ts-ignore we are not returning all the fields returned by the actual data field. We don't need to
        data: {
          connected: true,
        },
        loading: false,
      },
    ]);

    const spyUseAccount = jest.spyOn(wagmi, 'useAccount');
    spyUseAccount.mockReturnValue([
      {
        // @ts-ignore we are not returning all the fields returned by the actual data field. We don't need to
        data: {
          address: '0x123',
        },
        loading: false,
      },
    ]);
    render(
      <OnboardingLayout
        firstHeading="First heading"
        secondHeading="Second heading"
        bottomButtonText="Bottom content"
      >
        <p>Content</p>
      </OnboardingLayout>
    );
    expect(screen.getByText('First heading')).toBeInTheDocument();
    expect(screen.getByText('Second heading')).toBeInTheDocument();
    expect(screen.getByText('Content')).toBeInTheDocument();
    const btn = screen.getByRole('button', {
      name: 'Bottom content',
    });
    expect(btn).toBeInTheDocument();
    const mainContainer = screen.getByRole('main');
    expect(mainContainer).toBeInTheDocument();
    expect(mainContainer).toMatchSnapshot();
  });

  test('wallet not connected', () => {
    const spyUseConnect = jest.spyOn(wagmi, 'useConnect');
    spyUseConnect.mockReturnValue([
      {
        // @ts-ignore we are not returning all the fields returned by the actual data field. We don't need to
        data: {
          connected: false,
        },
        loading: false,
      },
    ]);

    const spyUseAccount = jest.spyOn(wagmi, 'useAccount');
    spyUseAccount.mockReturnValue([
      {
        // @ts-ignore we are not returning all the fields returned by the actual data field. We don't need to
        data: {},
        loading: false,
      },
    ]);
    render(
      <OnboardingLayout
        firstHeading="First heading"
        secondHeading="Second heading"
        bottomButtonText="Bottom content"
      >
        <p>Content</p>
      </OnboardingLayout>
    );
    const infoPrompt = screen.getByText('Please connect your wallet first');
    expect(infoPrompt).toBeInTheDocument();
  });
});
