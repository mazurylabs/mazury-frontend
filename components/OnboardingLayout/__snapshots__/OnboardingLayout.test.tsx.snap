// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardingLaout all sections are being rendered 1`] = `
<main
  class="mx-auto flex min-h-screen flex-col px-4 py-4 align-top sm:w-[530px]"
>
  <div
    class="absolute top-6 left-4 hidden w-full xl:flex"
  >
    <img
      alt="Mazury logo"
      class="rounded-full hover:cursor-pointer"
      height="40px"
      src="/waves.png"
      width="40px"
    />
  </div>
  <div
    class="xl:mt-20"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 16px; height: 16px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
    >
      <img
        alt="Back"
        class="hover:cursor-pointer"
        data-nimg="fixed"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
  </div>
  <h1
    class="mt-6 text-sm font-medium uppercase text-indigoGray-40"
  >
    First heading
  </h1>
  <h3
    class="mt-2 font-demi text-4xl text-indigoGray-90"
  >
    Second heading
  </h3>
  <p>
    Content
  </p>
  <div
    class="mx-auto mb-8 mt-8 w-full"
  >
    <button
      class="flex items-center justify-center gap-2 rounded-[8px] py-3 px-6 font-sans text-sm font-bold shadow-base disabled:cursor-not-allowed disabled:shadow-none bg-indigoGray-90  text-indigoGray-5 hover:bg-indigoGray-70 active:text-indigoGray-30  disabled:bg-indigoGray-30 w-full justify-center uppercase"
    >
      Bottom content
    </button>
  </div>
</main>
`;
